src/snowflake/snowpark/session.py

```diff
@@ -2519,6 +2526,15 @@ def table(
 
             _emit_ast: Whether to emit AST statements.
 
+            time_travel_mode: Time travel mode, either 'at' or 'before'. When specified,
+                returns a DataFrame with time travel query instead of a Table.
+                Exactly one of statement, offset, timestamp, or stream must be provided when time_travel_mode is set.
+            statement: Query ID for time travel.
+            offset: Negative integer representing seconds in the past for time travel.
+            timestamp: Timestamp string in 'YYYY-MM-DD HH:MM:SS' format or datetime object.
+            timezone: Timezone for timestamp conversion ('NTZ', 'LTZ', or 'TZ'). Defaults to 'NTZ'.
```

Comment:
The name timezone may be misleading. People may think the value is like `America/Los_Angeles`, `GMT-7`, etc.
I suggest `timestamp_type`. 

What's the behavior if the timestamp already has a timezone and the timestamp_type is NTZ?